### PR TITLE
[Merged by Bors] - chore: Do not import `MonoidWithZero` in `Data.Rat.Defs`

### PR DESCRIPTION
**Pull request by YaelDillies**
Move the `Ring` and `GroupWithZero` instances to a new file `Algebra.Ring.Rat`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Ring/Rat.lean [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+Authors: Johannes Hölzl, Mario Carneiro
+-/
+import Mathlib.Algebra.GroupWithZero.Units.Basic
+import Mathlib.Algebra.Ring.Int
+import Mathlib.Data.Rat.Defs
+
+#align_import data.rat.defs from "leanprover-community/mathlib"@"18a5306c091183ac90884daa9373fa3b178e8607"
+
+/-!
+# The rational numbers are a commutative ring for the Rational Numbers
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Ring/Rat.lean:13]: Say what now? :)

File: Mathlib/Data/Int/Cast/Lemmas.lean [hunk 2]
```diff
@@ -8,6 +8,10 @@ import Mathlib.Algebra.Order.Ring.Int
 import Mathlib.Algebra.Ring.Hom.Basic
 import Mathlib.Data.Nat.Cast.Commute
 import Mathlib.Data.Nat.Cast.Order
+import Mathlib.Algebra.Ring.Basic
+import Mathlib.Algebra.GroupWithZero.Units.Basic
+import Mathlib.Algebra.Order.Ring.Int
+import Mathlib.Data.Int.Cast.Defs
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Int/Cast/Lemmas.lean:14]: Sort please

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Algebra/Ring/Rat.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Ring/Rat.lean:13]:
```
# The rational numbers are a commutative ring
```

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Pull request closed without merging**